File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program172.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a java program which accept string from user and count number of small characters.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
#include<stdio.h>

using namespace std;

class StringInput
{
	public:

		char Arr[30];

		void Accept()
		{
			cout<<"Enter the String: ";
			scanf("%[^'\n']s",Arr);
		}

};

class StringDemo:public StringInput
{
	public:

		int CountSmall()
		{
			char *str = Arr;
			int iCnt=0;

			while(*str!='\0')
			{
				if(*str>='a' && *str<='z')
				{
					iCnt++;
				}
				str++;
			}

			return iCnt;
		}
};

int main()
{
	
	int iRet = 0;

	StringDemo *sObj = new StringDemo();
	sObj->Accept();	
	iRet = sObj->CountSmall();

	cout<<"Output: "<<iRet<<"\n";

	return 0;
}


/*
OUTPUT:

Enter the String: Vishal
Output: 5

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program172.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and count number of small characters.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class StringDemo
{
	public int CountCapital(String str)
	{
		int iCnt=0,iCount=0;
		while(iCnt!=str.length())
		{
			if(str.charAt(iCnt)>='a' && str.charAt(iCnt)<='z')
			{
				iCount++;
			}
			iCnt++;
		}
		
		return iCount;
	}
}


class Program2
{
	public static void main(String arg[])
	{
		int iRet = 0;

		Scanner sObj = new Scanner(System.in);
		System.out.println("Input: ");
		String str = sObj.nextLine();
	
		StringDemo sobj = new StringDemo();
		iRet = sobj.CountCapital(str);
	
		System.out.println(iRet+" ");
	}
}

/*
OUTPUT:

Input:
Vishal
6

*/
